src/routes/delegates/entities/schemas/__tests__/get-delegate.dto.schema.spec.ts

```diff
@@ -0,0 +1,99 @@
+import { GetDelegateDtoSchema } from '@/routes/delegates/entities/schemas/get-delegate.dto.schema';
+import { faker } from '@faker-js/faker';
+import { getAddress } from 'viem';
+import { ZodError } from 'zod';
+
+describe('GetDelegateDtoSchema', () => {
+  it('should validate a valid GetDelegateDto', () => {
+    const getDelegateDto = {
+      safe: getAddress(faker.finance.ethereumAddress()),
+      delegate: getAddress(faker.finance.ethereumAddress()),
+      delegator: getAddress(faker.finance.ethereumAddress()),
+      label: faker.lorem.word(),
+    };
+
+    const result = GetDelegateDtoSchema.safeParse(getDelegateDto);
+
+    expect(result.success).toBe(true);
+  });
+
+  it.each([
+    ['safe', getAddress(faker.finance.ethereumAddress())],
+    ['delegate', getAddress(faker.finance.ethereumAddress())],
+    ['delegator', getAddress(faker.finance.ethereumAddress())],
+    ['label', faker.lorem.word()],
+  ])('should validate with at least %s defined', (property, value) => {
+    const getDelegateDto = { [property]: value };
+
+    const result = GetDelegateDtoSchema.safeParse(getDelegateDto);
+
+    expect(result.success).toBe(true);
+  });
+
+  it('should not allow no properties', () => {
+    const getDelegateDto = {};
+
+    const result = GetDelegateDtoSchema.safeParse(getDelegateDto);
+
+    expect(!result.success && result.error).toStrictEqual(
+      new ZodError([
+        {
+          code: 'custom',
+          message: 'At least one property is required',
+          path: [],
+        },
+      ]),
+    );
+  });
+
+  it.each([['safe' as const], ['delegate' as const], ['delegator' as const]])(
```

Comment:
Are the tuples required? Wouldn't the following work:

```typescript
it.each(['safe' as const, 'delegate' as const, 'delegator' as const])
```

Reply:
Removed in 87e895a402b63bbdf461528f7f5507466e315d6f.